pkg/plugins/layout.go

```diff
@@ -356,7 +368,7 @@ func (dev Disk) FindPartitionDevice(partNum int, console Console) (string, error
 	}
 	match := re.FindStringSubmatch(out)
```

Comment:
A doubt about using the FindStringSubmatch:

consider there are:
/dev/sda1
/dev/sda11

lsblk is returned as
/dev/sda11
/dev/sda1

when look for  /dev/sda1,  is it OK?

lsblk has an option to sort list:
 ```
      -x, --sort column
              Sort  output  lines by column. This option enables --list output format by default. 
               It is possible to use the option --tree to force tree-like output and than
              the tree branches are sorted by the column.
```

or, there is only a 
/dev/sda11

but looking for /dev/sda1, is it returning OK?

Maybe an exact match check is necessary, even though it is in a low possible rate for such corner cases.

Reply:
that is a good point, I will follow up with an extra patch to check for that afterwards!

Reply:
we cna actually do better than that and just try to get the full partition directly from lsblk, after all we have the device and the partition number, so we can construct something like:

```
func (dev Disk) FindPartitionDevice(partNum int, console Console) (string, error) {
	out, err := console.Run(fmt.Sprintf("lsblk -ltnpo name,type %s%d", dev.Device, partNum))
	if err != nil {
		return "", errors.New(fmt.Sprintf("Could not find in device %s partition number %d: %s", dev.Device, partNum, out))
	}
	return fmt.Sprintf("%s%d", dev.Device, partNum), nil
```

@davidcassany do you think this could work?

Reply:
PR sent with this, was even more straightforward than expected: https://github.com/mudler/yip/pull/28